src/validation.cpp

```diff
@@ -4729,6 +4728,7 @@ bool LoadExternalBlockFile(const CChainParams& chainparams, FILE* fileIn, FlatFi
 
 void CChainState::CheckBlockIndex(const Consensus::Params& consensusParams)
 {
+    static bool fCheckBlockIndex = gArgs.GetBoolArg("-checkblockindex", chainparams.DefaultConsistencyChecks());
```

Comment:
```
validation.cpp:4731:73: error: use of undeclared identifier 'chainparams'

    static bool fCheckBlockIndex = gArgs.GetBoolArg("-checkblockindex", chainparams.DefaultConsistencyChecks());
```
`chainparams` is not available in the scope, you should give a `CChainParams` argument in the function (like in the other functions) or use a global variable

Reply:
Thank you for your review @emilengler. I've updated the CheckBlockIndex function header to take all of `chainparams` instead of only the `consensusParams` subset.

Reply:
Instead of parsing command line arguments every time this function is called, what about adding this as a member variable that is only parsed once (e.g. at the time of construction).

Reply:
Sorry, I was wrong this was parsed on every call (missed the `static`), however I'd still argue that this should be parsed at the time of construction (during initialization), so that parse errors can be returned to the caller.

Reply:
(See also the reply by @ryanofsky which says exactly the same)